zilliqa-macros/src/test.rs

```diff
@@ -22,33 +22,45 @@ pub(crate) fn test_macro(_args: TokenStream, item: TokenStream) -> TokenStream {
             // The original test function
             #input
 
+            let run_sequentially = std::env::var_os("ZQ_RUN_SEQUENTIALLY").is_some();
```

Comment:
convenient for when it will fail when running many seeds.

I actually did have it failing on multiple seed, but I have yet to encounter the situation where it always fails on the same seed, so there is some non-determanism in the test I believe.

Reply:
Also, the println always prints at the end, its a todo to interleave it with the tracing framework somehow

Reply:
I've seen a similar issue, where it is hard to match the `ZQ_TEST_SEED` messages with the runs that actually failed. This is annoying if only one seed failed and you get 100 possible seeds that caused it.

e.g.
https://github.com/Zilliqa/zq2/actions/runs/6670165505/job/18129486734#step:9:156

I will see if we can fix this more elegantly by controlling how we print things in the `JoinSet` stuff of the test harness...

Reply:
#534 , sorry it took a while.